enclave/src/lib.rs

```diff
@@ -102,9 +103,22 @@ pub mod rpc;
 pub mod tls_ra;
 pub mod top_pool;
 
+#[cfg(feature = "test")]
 pub mod test;
+
+#[cfg(feature = "test")]
 pub mod tests;
 
+#[cfg(not(feature = "test"))]
+use sgx_types::size_t;
+
+// this is a 'dummy' for production mode
+#[cfg(not(feature = "test"))]
+#[no_mangle]
+pub extern "C" fn test_main_entrance() -> size_t {
+	0
```

Comment:
Ah, I missed this yesterday. How about putting an `unreachable!("Tests are not available when compiled in production mode.")`